src/Streamly/Coreutils/Mkdir.hs

```diff
@@ -0,0 +1,21 @@
+-- |
+-- Module      : Streamly.Coreutils.Mkdir
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Create the Directory(ies), if they do not already exist.
+
+module Streamly.Coreutils.Mkdir where
+
+import System.Directory (createDirectory, createDirectoryIfMissing)
+
+data MkdirOptions = NO_PARENT | PARENT
```

Comment:
Why is this not a Switch with a default value? We are exposing NO_PARENT and PARENT here which is not how we do this for any other utilities.